README
NAME

    Pod::Weaver::Section::Support - Add a SUPPORT section to your POD

VERSION

      This document describes v1.006 of Pod::Weaver::Section::Support - released October 25, 2014 as part of Pod-Weaver-Section-Support.

DESCRIPTION

    This section plugin will produce a hunk of pod that lists the various
    ways to get support for this module. It will do this only if it is
    being built with Dist::Zilla because it needs the data from the dzil
    object.

    If you have Dist::Zilla::Plugin::Repository enabled in your dist.ini,
    be sure to check the repository_link attribute!

    This is added ONLY to the main module's POD, because it would be a
    waste of space to add it to all modules in the dist.

    For an example of what the hunk looks like, look at the "SUPPORT"
    section in this POD :)

ATTRIBUTES

 all_modules

    Enable this if you want to add the SUPPORT section to all the modules
    in a dist, not only the main one.

    The default is false.

 perldoc

    Specify if you want the paragraph explaining about perldoc to be
    displayed or not.

    The default is true.

 bugs

    Specify the bugtracker you want to use. You can use the CPAN RT tracker
    or your own, specified in the metadata.

    Valid options are: "rt", "metadata", or "none"

    If you pick the "rt" option, this module will generate a predefined
    block of text explaining how to use the RT system.

    If you pick the "metadata" option, this module will check the
    Dist::Zilla metadata for the bugtracker to display. Be sure to verify
    that your metadata contains both 'web' and 'mailto' keys if you want to
    use them in the content!

    The default is "rt".

 bugs_content

    Specify the content for the bugs section.

    Please put the "{EMAIL}" and "{WEB}" placeholders somewhere!

    The default is a sufficient explanation (see "SUPPORT").

 websites

    Specify what website links you want to see. This is an array, and you
    can pick any combination. You can also specify it as a comma-delimited
    string. The ordering of the options are important, as they are
    reflected in the final POD.

    Valid options are: "none", "metacpan", "search", "rt", "anno",
    "ratings", "forum", "kwalitee", "testers", "testmatrix", "deps" and
    "all".

    The default is "all".

            # Where the links go to:
            metacpan        - http://metacpan.org/release/$dist
            search          - http://search.cpan.org/dist/$dist
            rt              - https://rt.cpan.org/Public/Dist/Display.html?Name=$dist
            anno            - http://annocpan.org/dist/$dist
            ratings         - http://cpanratings.perl.org/d/$dist
            forum           - http://cpanforum.com/dist/$dist
            kwalitee        - http://cpants.perl.org/dist/overview/$dist
            testers         - http://cpantesters.org/distro/$first_char/$dist
            testmatrix      - http://matrix.cpantesters.org/?dist=$dist
            deps            - http://deps.cpantesters.org/?module=$module
    
            # in weaver.ini
            [Support]
            websites = search
            websites = forum
            websites = testers , testmatrix

    P.S. If you know other websites that I should include here, please let
    me know!

 websites_content

    Specify the content to be displayed before the website list.

    The default is a sufficient explanation (see "SUPPORT").

 irc

    Specify an IRC server/channel/nick for online support. You can specify
    as many networks/channels as you want. The ordering of the options are
    important, as they are reflected in the final POD.

    You specify a network, then a list of channels/nicks to ask for
    support. There are two ways to format the string:

            servername.com,#room,nick
            irc://servername.com/#room

    The default is none.

            # in weaver.ini
            [Support]
            irc = irc.home.org, #support, supportbot
            irc = irc.acme.com, #acmecorp, #acmehelp, #acmenewbies

    You can also add the irc information in the distribution metadata via
    Dist::Zilla::Plugin::Metadata. Valid keys are 'x_irc' or 'IRC' but you
    have to use the irc:// format to retain compatibility with the rest of
    the ecosystem.

            # in dist.ini
            [Metadata]
            x_irc = irc://irc.perl.org/#perl

 irc_content

    Specify the content to be displayed before the irc network/channel
    list.

    The default is a sufficient explanation (see "SUPPORT").

 repository_link

    Specify which url to use when composing the external link. The value
    corresponds to the repository meta resources (for dzil v3 with CPAN
    Meta v2).

    Valid options are: "url", "web", "both", or "none".

    "both" will include links to both the "url" and "web" in separate POD
    paragraphs.

    "none" will skip the repository item entirely.

    The default is "both".

    An error will be thrown if a specified link is not found because if you
    said that you wanted it you probably expect it to be there.

 repository_content

    Specify the content to be displayed before the link to the source code
    repository.

    The default is a sufficient explanation (see "SUPPORT").

 email

    Specify an email address here so users can contact you directly for
    help.

    If you supply a string without '@' in it, we assume it is a PAUSE id
    and mangle it into 'USER at cpan.org'.

    The default is none.

 email_content

    Specify the content for the email section.

    Please put the "{EMAIL}" placeholder somewhere!

    The default is a sufficient explanation ( see "SUPPORT").

SUPPORT

 Perldoc

    You can find documentation for this module with the perldoc command.

      perldoc Pod::Weaver::Section::Support

 Websites

    The following websites have more information about this module, and may
    be of help to you. As always, in addition to those websites please use
    your favorite search engine to discover more resources.

      * MetaCPAN

      A modern, open-source CPAN search engine, useful to view POD in HTML
      format.

      http://metacpan.org/release/Pod-Weaver-Section-Support

      * Search CPAN

      The default CPAN search engine, useful to view POD in HTML format.

      http://search.cpan.org/dist/Pod-Weaver-Section-Support

      * RT: CPAN's Bug Tracker

      The RT ( Request Tracker ) website is the default bug/issue tracking
      system for CPAN.

      http://rt.cpan.org/NoAuth/Bugs.html?Dist=Pod-Weaver-Section-Support

      * AnnoCPAN

      The AnnoCPAN is a website that allows community annotations of Perl
      module documentation.

      http://annocpan.org/dist/Pod-Weaver-Section-Support

      * CPAN Ratings

      The CPAN Ratings is a website that allows community ratings and
      reviews of Perl modules.

      http://cpanratings.perl.org/d/Pod-Weaver-Section-Support

      * CPAN Forum

      The CPAN Forum is a web forum for discussing Perl modules.

      http://cpanforum.com/dist/Pod-Weaver-Section-Support

      * CPANTS

      The CPANTS is a website that analyzes the Kwalitee ( code metrics )
      of a distribution.

      http://cpants.cpanauthors.org/dist/overview/Pod-Weaver-Section-Suppor
      t

      * CPAN Testers

      The CPAN Testers is a network of smokers who run automated tests on
      uploaded CPAN distributions.

      http://www.cpantesters.org/distro/P/Pod-Weaver-Section-Support

      * CPAN Testers Matrix

      The CPAN Testers Matrix is a website that provides a visual overview
      of the test results for a distribution on various Perls/platforms.

      http://matrix.cpantesters.org/?dist=Pod-Weaver-Section-Support

      * CPAN Testers Dependencies

      The CPAN Testers Dependencies is a website that shows a chart of the
      test results of all dependencies for a distribution.

      http://deps.cpantesters.org/?module=Pod::Weaver::Section::Support

 Email

    You can email the author of this module at APOCAL at cpan.org asking
    for help with any problems you have.

 Internet Relay Chat

    You can get live help by using IRC ( Internet Relay Chat ). If you
    don't know what IRC is, please read this excellent guide:
    http://en.wikipedia.org/wiki/Internet_Relay_Chat. Please be courteous
    and patient when talking to us, as we might be busy or sleeping! You
    can join those networks/channels and get help:

      * irc.perl.org

      You can connect to the server at 'irc.perl.org' and join this
      channel: #perl-help then talk to this person for help: Apocalypse.

      * irc.freenode.net

      You can connect to the server at 'irc.freenode.net' and join this
      channel: #perl then talk to this person for help: Apocal.

      * irc.efnet.org

      You can connect to the server at 'irc.efnet.org' and join this
      channel: #perl then talk to this person for help: Ap0cal.

 Bugs / Feature Requests

    Please report any bugs or feature requests by email to
    bug-pod-weaver-section-support at rt.cpan.org, or through the web
    interface at
    http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Pod-Weaver-Section-Suppo
    rt. You will be automatically notified of any progress on the request
    by the system.

 Source Code

    The code is open to the world, and available for you to hack on. Please
    feel free to browse it and play with it, or whatever. If you want to
    contribute patches, please send me a diff or prod me to pull from your
    repository :)

    https://github.com/apocalypse/perl-pod-weaver-section-support

      git clone git://github.com/apocalypse/perl-pod-weaver-section-support.git

AUTHOR

    Apocalypse <APOCAL@cpan.org>

 CONTRIBUTORS

      * Alex Peters <lxp@cpan.org>

      * Kent Fredric <kentfredric@gmail.com>

      * Randy Stauner <randy@magnificent-tears.com>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2014 by Apocalypse.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.

    The full text of the license can be found in the LICENSE file included
    with this distribution.

DISCLAIMER OF WARRANTY

    THERE IS NO WARRANTY FOR THE PROGRAM, TO THE EXTENT PERMITTED BY
    APPLICABLE LAW. EXCEPT WHEN OTHERWISE STATED IN WRITING THE COPYRIGHT
    HOLDERS AND/OR OTHER PARTIES PROVIDE THE PROGRAM "AS IS" WITHOUT
    WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING, BUT NOT
    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
    PARTICULAR PURPOSE. THE ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE
    OF THE PROGRAM IS WITH YOU. SHOULD THE PROGRAM PROVE DEFECTIVE, YOU
    ASSUME THE COST OF ALL NECESSARY SERVICING, REPAIR OR CORRECTION.

    IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
    WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MODIFIES AND/OR
    CONVEYS THE PROGRAM AS PERMITTED ABOVE, BE LIABLE TO YOU FOR DAMAGES,
    INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES
    ARISING OUT OF THE USE OR INABILITY TO USE THE PROGRAM (INCLUDING BUT
    NOT LIMITED TO LOSS OF DATA OR DATA BEING RENDERED INACCURATE OR LOSSES
    SUSTAINED BY YOU OR THIRD PARTIES OR A FAILURE OF THE PROGRAM TO
    OPERATE WITH ANY OTHER PROGRAMS), EVEN IF SUCH HOLDER OR OTHER PARTY
    HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.



Changes
Revision history for Pod::Weaver::Section::Support

1.006	2014-10-26 05:12:19 UTC

	Documentation tweaks
	Resolve RT#78450 - retrieve IRC information from the dzil metadata if possible ( thanks BBYRD! )

1.005	2012-01-03 00:48:32 UTC

	Add the MetaCPAN website to the website list ( thanks RWSTAUNER! )
	Tweak the logging system a bit ( thanks KENTNL! )
	Fix some internal spelling mistakes ( thanks RWSTAUNER! )
	Resolve RT#69033 - logging tweaks ( thanks RJBS! )

1.004	2011-04-21 21:48:17 UTC

	Tweaked the output of the websites so it also includes a descriptive text of the link
	Add the deps.cpantesters.org website ( deps ) to the website list

1.003	2011-02-24 17:03:11 UTC

	Made sure the bugs_content and the metadata match up, otherwise it will throw an error! ( thanks XENO! )
	Changed the *_content attributes to arrayrefs so you can sanely use them in a weaver.ini ( thanks XENO! )
	The email attribute didn't add C<> to the address if you specified your own ( thanks XENO! )

1.002	2011-02-02 22:55:07 UTC

	Fix a few typos in the POD
	Add email section to contact the author
	Make the perldoc section a proper header
	Tweaked the wording of the bugs section ( thanks DAGOLDEN! )
	Add attribute to specify the bugs content ( thanks DAGOLDEN! )
	Add attribute to specify the website content

1.001	2010-12-08 07:35:41 UTC

	Add attributes to control the repository section ( thanks RWSTAUNER! )
	Reorganized the layout of the generated POD
	Add attribute to control the website links ( thanks GETTY! )
	Add attributes for new irc section ( thanks GETTY! )
	Add attribute to control the bugs section ( thanks GETTY! )
	Add attribute to enable/disable perldoc section ( thanks GETTY! )
	Add attribute to enable/disable SUPPORT section for all modules ( thanks GETTY! )

1.000	2010-05-30 18:03:49 UTC

	Work with CPAN Meta v2 by recognizing the repository
	Make the repository hunk of POD optional

0.010	2010-05-28 17:41:54 UTC

	fixed the way the stopwords were added to the POD

0.009	2010-05-13 01:41:59 UTC

	another fake release

0.008	2010-05-12 04:50:01 UTC

	fake release

0.007	2010-05-11 21:12:28 UTC

	fake release

0.006	2010-05-11 17:05:55 UTC

	Tweaked the POD a little, and make another release...

0.005	2010-05-11 06:26:29 UTC

	real CPAN release, ha!

0.004	2010-05-11 06:25:00 UTC

	add example weaver.ini

0.003	2010-05-11 06:16:15 UTC

	yet another test release

0.002	2010-05-11 06:13:52 UTC

	test release

0.001	2010-05-11 03:29:40 UTC

	initial release


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.021.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.28"
  },
  "configure_requires" => {
    "Module::Build" => "0.28"
  },
  "dist_abstract" => "Add a SUPPORT section to your POD",
  "dist_author" => [
    "Apocalypse <APOCAL\@cpan.org>"
  ],
  "dist_name" => "Pod-Weaver-Section-Support",
  "dist_version" => "1.006",
  "license" => "perl",
  "module_name" => "Pod::Weaver::Section::Support",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Moose" => "1.03",
    "Moose::Autobox" => "0.10",
    "Moose::Util::TypeConstraints" => "1.01",
    "Pod::Weaver::Role::Section" => "3.100710",
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.88",
    "perl" => "5.006"
  }
);


my %fallback_build_requires = (
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Module::Build" => "0.28",
  "Test::More" => "0.88",
  "perl" => "5.006"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.021.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Add a SUPPORT section to your POD",
  "AUTHOR" => "Apocalypse <APOCAL\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Module::Build" => "0.28"
  },
  "CONFIGURE_REQUIRES" => {
    "Module::Build" => "0.28"
  },
  "DISTNAME" => "Pod-Weaver-Section-Support",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Pod::Weaver::Section::Support",
  "PREREQ_PM" => {
    "Moose" => "1.03",
    "Moose::Autobox" => "0.10",
    "Moose::Util::TypeConstraints" => "1.01",
    "Pod::Weaver::Role::Section" => "3.100710",
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.88"
  },
  "VERSION" => "1.006",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Module::Build" => "0.28",
  "Moose" => "1.03",
  "Moose::Autobox" => "0.10",
  "Moose::Util::TypeConstraints" => "1.01",
  "Pod::Weaver::Role::Section" => "3.100710",
  "Test::More" => "0.88",
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



